src/AutoMapper/TypeMap.cs

```diff
@@ -284,7 +284,8 @@ public void Seal(IConfigurationProvider configurationProvider, Stack<TypeMap> ty
             _orderedPropertyMaps =
                 _propertyMaps
                     .Union(_inheritedMaps)
-                    .OrderBy(map => map.MappingOrder).ToArray();
+                    .OrderBy(map => map.MappingOrder ?? int.MaxValue).ThenBy(map => map.DestinationProperty?.Name)
+                    .ToArray();
```

Comment:
The issue is about EF and ProjectTo, Map works just fine, so I don't see any reason to touch Map. Also the order by name is too obvious and people might be tempted to rely on it. I would order by DestinationProperty.MetadaToken. That should be consistent and faster.

Reply:
The reason for touching it where I did was that MappingOrder was already done the same place, and to avoid having the same logic happen different places I put it there to avoid confusion from for others. I can't really see the reason why MappingOrder should be here and not default mapping?

As for DestinationProperty it doesn't have any MetadataToken?

Reply:
Yes, I see it's not easy to get a hold of that MetadataToken. Then I guess there is nothing else to sort on. DestinationProperty will always be set, so that null check is useless. I don't think MappingOrder makes much sense for ProjectTo, so I'd rather have a built in ordering, but @jbogard  might disagree.

Reply:
People may use MappingOrder already, so I think ignoring it would be a bad idea for them.

Reply:
MappingOrder is irrelevant for ProjectTo though. I think the fix is just to order by name during projection. Can you write an integration test that shows the failure? I'd really rather see a LINQ query that uses a union that actually shows a failure.

Reply:
Ok! I'll move the code to the projection! An integration test will also be made!